# Assignment Nov 2019 Machine Learning and Statistics - GMIT 

## Introduction and Overview

This assignment concerns the well known Boston House Prices Dataset. The dataset was created from data taken by the US Census Service in 1978. It contains information about house pricing and other associated characteristics in Boston. More information can be seen in the following Print and Describe commands below. 

This assignment consists of the following sections:

1. Description - Descriptive statistics and plots to describe the dataset.
2. Infer - Inferential statistics to analyse whether there is a significant difference in median house prices between those that are along the Charles river and those that are not.
3. Predict - Create a neural network that can predict the median house price based on other variables in the dataset. 

In [11]:
# importing modules/libraries 

from sklearn.datasets import load_boston
import numpy as np
import scipy as sp
import pandas as pd

In [4]:
boston = load_boston() # load the dataset built into sklearn

In [5]:
print (boston.data.shape) # print the shape of it, what is it, what's in it

(506, 13)


So it looks like the dataset has 506 rows and 13 columns.

In [6]:
print(boston.DESCR)

.. _boston_dataset:

Boston house prices dataset
---------------------------

**Data Set Characteristics:**  

    :Number of Instances: 506 

    :Number of Attributes: 13 numeric/categorical predictive. Median Value (attribute 14) is usually the target.

    :Attribute Information (in order):
        - CRIM     per capita crime rate by town
        - ZN       proportion of residential land zoned for lots over 25,000 sq.ft.
        - INDUS    proportion of non-retail business acres per town
        - CHAS     Charles River dummy variable (= 1 if tract bounds river; 0 otherwise)
        - NOX      nitric oxides concentration (parts per 10 million)
        - RM       average number of rooms per dwelling
        - AGE      proportion of owner-occupied units built prior to 1940
        - DIS      weighted distances to five Boston employment centres
        - RAD      index of accessibility to radial highways
        - TAX      full-value property-tax rate per $10,000
        - PTRATIO  pu

The dataset has 13 columns but in the DESCR output above there are 14 attributes. It mentions that MEDV (median value of houses) is usually the target. Opening the above link to the dataset gives a number of files. They are Housing names and Housing data. Opening these in a text editor doesn't simplify anything.  

In [8]:
print(boston.feature_names) 
# ouput the feature names to check/understand what is in the columns since there are 14 attributs and 13 are listed 

['CRIM' 'ZN' 'INDUS' 'CHAS' 'NOX' 'RM' 'AGE' 'DIS' 'RAD' 'TAX' 'PTRATIO'
 'B' 'LSTAT']


In [10]:
print (boston.keys()) # what else is in the dataset?

dict_keys(['data', 'target', 'feature_names', 'DESCR', 'filename'])


The following print commands can be used to view the contents of each of the keys 

In [22]:
#print (boston.data)
#print (boston.target)
#print (boston.feature_names)
#print (boston.filename)

In [12]:
bos = pd.DataFrame(boston.data) # pass to pandas because it's still not 100% clear what is in this dataset

In [23]:
print (bos.head(2)) # ok but what does this mean, there are only 13 here. Where is the price/ what are these numbers?

        0     1     2    3      4      5     6       7    8      9     10  \
0  0.00632  18.0  2.31  0.0  0.538  6.575  65.2  4.0900  1.0  296.0  15.3   
1  0.02731   0.0  7.07  0.0  0.469  6.421  78.9  4.9671  2.0  242.0  17.8   

      11    12  
0  396.9  4.98  
1  396.9  9.14  
